src/Core/Security/Vault/Domain/Model/NewVaultConfiguration.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/*
+ * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Security\Vault\Domain\Model;
+
+use Centreon\Domain\Common\Assertion\{Assertion, AssertionException};
+
+/**
+ * This class represents vault configuration being created.
+ */
+class NewVaultConfiguration
+{
+    public const MAX_LENGTH = 255;
+    public const MIN_PORT_VALUE = 1;
+    public const MAX_PORT_VALUE = 65535;
+    public const TYPE_HASHICORP = 'hashicorp';
```

Comment:
You should put an integer that represent the unique identifier

Reply:
Done